src/history.rs

```diff
@@ -219,6 +219,22 @@ impl Queue {
     }
 }
 
+#[derive(Debug)]
+pub(crate) enum HistoryError {
+    NoBlocksAboveThreshold,
+    RelativeThresholdIsNan,
+}
+
+impl fmt::Display for HistoryError {
```

Comment:
```suggestion
impl std::error::Error for HistoryError {}

impl fmt::Display for HistoryError {
```

too. We don't need it right now but it's cleaner to have error enums already in the correct shape

Reply:
Sorry, I was somehow under the wrong impression that `std::error::Error` is an auto-trait when Debug/Display are defined.